src/services/navigationBar.ts

```diff
@@ -2,856 +2,552 @@
-
-            //        childNodes.push(node);
-            //    }
-            //    else if (node.kind === SyntaxKind.VariableStatement) {
-            //        childNodes.push.apply(childNodes, (<VariableStatement>node).declarations);
-            //    }
-            // }
-            forEach(nodes, visit);
-            return sortNodes(childNodes);
-        }
-
-        function getTopLevelNodes(node: SourceFile): Node[] {
-            const topLevelNodes: Node[] = [];
-            topLevelNodes.push(node);
-
-            addTopLevelNodes(node.statements, topLevelNodes);
-
-            return topLevelNodes;
-        }
+    export function getNavigationBarItems(sourceFile: SourceFile): NavigationBarItem[] {
+        const root = createNavNode(undefined, sourceFile);
+        return map(topLevelItems(root), convertToTopLevelItem);
+    }
 
-        function sortNodes(nodes: Node[]): Node[] {
-            return nodes.slice(0).sort((n1: Declaration, n2: Declaration) => {
-                if (n1.name && n2.name) {
-                    return localeCompareFix(getPropertyNameForPropertyNameNode(n1.name), getPropertyNameForPropertyNameNode(n2.name));
-                }
-                else if (n1.name) {
-                    return 1;
-                }
-                else if (n2.name) {
-                    return -1;
-                }
-                else {
-                    return n1.kind - n2.kind;
-                }
-            });
+    /**
+     * Represents a navBar item and its children.
+     * The returned NavigationBarItem is more complicated and doesn't include 'parent', so we use these to do work before converting.
+     */
+    interface NavNode {
+        node: Node;
+        additionalNodes: Node[]; // May be missing
+        parent: NavNode; // Missing for root decl
+        children: NavNode[];
+        indent: number; // # of parents
+    }
+    function navKind(n: NavNode): SyntaxKind {
+        return n.node.kind;
+    }
+    function navModifiers(n: NavNode): string {
+        return getNodeModifiers(n.node);
+    }
 
-            // node 0.10 treats "a" as greater than "B".
-            // For consistency, sort alphabetically, falling back to which is lower-case.
-            function localeCompareFix(a: string, b: string) {
-                const cmp = a.toLowerCase().localeCompare(b.toLowerCase());
-                if (cmp !== 0)
-                    return cmp;
-                // Return the *opposite* of the `<` operator, which works the same in node 0.10 and 6.0.
-                return a < b ? 1 : a > b ? -1 : 0;
-            }
+    /** Creates a child node and adds it to parent. */
+    function createNavNode(parent: NavNode, node: Node): NavNode {
+        const navNode: NavNode = {
+            node,
+            additionalNodes: undefined,
+            parent,
+            children: [],
+            indent:
+            parent ? parent.indent + 1 : 0

```

Comment:
This conditional being on the line following the key but at the same indentation is confusing - it messes up the line of keys down the left. Either put it on the above line or indent it, to make it visually distinct. (Note to self: Add a lint rule for this)